File: solutions_python/strings/14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python
#
# [14] Longest Common Prefix
#

# @lc code=start
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        res = ""
        if not strs:
            return res
        
        shortest = strs[0]
        for str in strs:
            if len(str) < len(shortest):
                shortest = str
        
        for i in range(len(shortest)):
            curr = shortest[i]
            for str in strs:
                if str[i] != curr:
                    return res
            res += curr
        return res
        
        
# @lc code=end

File: solutions_java/arrays/14.longest-common-prefix.java
/*
 * @lc app=leetcode id=14 lang=java
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int resIdx = 0;
        
        Arrays.sort(strs);
        for (int i=0; i<strs[0].length(); i++) {
            if (strs[0].charAt(i) == strs[strs.length-1].charAt(i)) {
                resIdx++;
            } else {
                break;
            }
        }
        return strs[0].substring(0,resIdx);
        
    }
}
// @lc code=end

